### PR TITLE
Improve pymysql.converters stubs

### PR DESCRIPTION
**Pull request by JelleZijlstra**
I spent far too much time being confused about why pyanalyze thought `pymysql.converters.escape_dict` only takes two arguments.

I rewrote the stubs from scratch using the implementation: https://github.com/PyMySQL/PyMySQL/blob/main/pymysql/converters.py.

The "charset" argument is ignored as far as I can tell; it gets passed to other functions but no function actually uses it.

### CODE REVIEW

File: stubs/PyMySQL/pymysql/converters.pyi [hunk 1]
```diff
@@ -1,47 +1,39 @@
...
+def escape_bytes_prefixed(value: bytes, mapping: _EscaperMapping = ...) -> str: ...
+def escape_bytes(value: bytes, mapping: _EscaperMapping = ...) -> str: ...
+def escape_str(value: str, mapping: _EscaperMapping = ...) -> str: ...
+def escape_None(value: None, mapping: _EscaperMapping = ...) -> str: ...
+def escape_timedelta(obj: datetime.timedelta, mapping: _EscaperMapping = ...) -> str: ...
+def escape_time(obj: datetime.time, mapping: _EscaperMapping = ...) -> str: ...
+def escape_datetime(obj: datetime.datetime, mapping: _EscaperMapping = ...) -> str: ...
+def escape_date(obj: datetime.date, mapping: _EscaperMapping = ...) -> str: ...
+def escape_struct_time(obj: time.struct_time, mapping: _EscaperMapping = ...) -> str: ...
+def Decimal2Literal(o: Decimal, d: _EscaperMapping) -> str: ...
```
Comment (srittau) [inline stubs/PyMySQL/pymysql/converters.pyi:27]: This should just copy the definition of `builtins.format()` (minus the default):

```python
def format(__value: object, __format_spec: str = ...) -> str: ...
```

File: stubs/PyMySQL/pymysql/converters.pyi [hunk 2]
```diff
@@ -1,47 +1,39 @@
...
+def convert_datetime(obj: str | bytes) -> datetime.datetime | str: ...
+def convert_timedelta(obj: str | bytes) -> datetime.timedelta | str: ...
+def convert_time(obj: str | bytes) -> datetime.time | str: ...
+def convert_date(obj: str | bytes) -> datetime.date | str: ...
+def through(x: _T) -> _T: ...
+
+convert_bit = through
+
+encoders: dict[type[object], Callable[..., str]]
+decoders: dict[int, Callable[[str], Any]]
```
Comment (srittau) [inline stubs/PyMySQL/pymysql/converters.pyi:37]:
```suggestion
decoders: dict[int, Callable[[str | bytes], Any]]
```

**Review by srittau — CHANGES_REQUESTED** [review]
Looks mostly good, two remarks below.

**Comment by srittau** [comment]
Dropping `python2 = true` from `METADATA.toml` should fix CI. Python 2 is not listed as supported at `https://pypi.org/project/PyMySQL/`.

File: stubs/PyMySQL/pymysql/converters.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stubs/PyMySQL/pymysql/converters.pyi:27]: It actually ignores the second argument. I'll type it as `object`.

And the first argument can only be an object that accepts `"f"` as a format specifier. I don't know what other types that would include, but clearly it's meant for Decimal, so I'd prefer to keep that as the annotation.

**Review by srittau — APPROVED** [review]

File: stubs/PyMySQL/pymysql/converters.pyi [hunk 3]
```diff
@@ -1,47 +1,39 @@
...
+def convert_timedelta(obj: str | bytes) -> datetime.timedelta | str: ...
+def convert_time(obj: str | bytes) -> datetime.time | str: ...
+def convert_date(obj: str | bytes) -> datetime.date | str: ...
+def through(x: _T) -> _T: ...
+
+convert_bit = through
+
+encoders: dict[type[object], Callable[..., str]]
+decoders: dict[int, Callable[[str | bytes], Any]]
+conversions: dict[type[object] | int, Callable[..., Any]]
```
Comment (srittau) [inline stubs/PyMySQL/pymysql/converters.pyi:38]: `type` -> `Type` (mypy 😞)

Comment (JelleZijlstra) [reply stubs/PyMySQL/pymysql/converters.pyi:38]:
```suggestion
encoders: dict[type[object], Callable[..., str]]
decoders: dict[int, Callable[[str | bytes], Any]]
conversions: dict[Type[object] | int, Callable[..., Any]]
```

**Pull request merged**